broke-broker/src/main/java/org/dsngroup/broke/broker/storage/ServerSessionCollection.java

```diff
@@ -0,0 +1,62 @@
+package org.dsngroup.broke.broker.storage;
+
+import io.netty.handler.codec.mqtt.MqttMessage;
+
+import java.util.HashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * The singleton class ServerSessionCollection
+ * Contain all the Server Sessions for all clients
+ * */
+public class ServerSessionCollection {
+
+    // Use volatile to guarantee all session user threads gets the well initialized instance.
+    private volatile static ServerSessionCollection serverSessionCollection;
+
+    private HashMap<String, ServerSession> serverSessionCollectionMap;
+
+    /**
+     * Getter method of the server session collection
+     * */
+    public static ServerSessionCollection getServerSessionCollection() {
+        if(serverSessionCollection==null) {
+            synchronized (ServerSessionCollection.class) {
+                if(serverSessionCollection==null) {
+                    serverSessionCollection = new ServerSessionCollection();
+                }
+            }
+        }
+        return serverSessionCollection;
+    }
+
+    /**
+     * Function for clients to get the session upon CONNECT
+     * If the cleanSession flag is false, use existed session. If not existed, create a new one.
+     * If the cleanSession flag is true, replace the old session with a new one.
+     * See "Clean Session" flag in Mqtt Specification
+     * @param clientId client ID
+     * @param cleanSession Clean Session Flag
+     * */
+    public ServerSession getSession(String clientId, boolean cleanSession) {
+
+        if (cleanSession) {
+            if(serverSessionCollectionMap.containsKey(clientId))
+                serverSessionCollectionMap.remove(clientId);
+            serverSessionCollectionMap.put(clientId, new ServerSession(clientId));
+        } else {
+            if(!serverSessionCollectionMap.containsKey(clientId))
+                serverSessionCollectionMap.put(clientId, new ServerSession(clientId));
+        }
+        return serverSessionCollectionMap.get(clientId);
+
+    }
```

Comment:
Sync?

Reply:
Ok, but further. How about integrate `isConnect` here instead of `processor`?

Reply:
I think "isConnect" is a "connection-wise" status, as same as the protocol processor. But the Session is a "client-wise" status, which is specified using clientId. That's why I placed the isConnect in protocol processor.

Reply:
Ok, it depends on you. Additional ask, can we check is connect if a session has been kept?

Reply:
I haven't done this yet. I think I'll add a connection status keeper in the ServerSessionCollection, which is used to identify whether a session is in use. This status keeper will be maintained and accessed by the protocol processors.

Reply:
Oh yes, this is exactly what I mean, you can mark it as TODO currently.

Reply:
I did a prototype in the latest commit. Please take a look!